tests/DeferredTest.php

```diff
@@ -54,6 +54,7 @@ public function shouldRejectWithoutCreatingGarbageCyclesIfCancellerHoldsReferenc
         gc_collect_cycles();
         gc_collect_cycles(); // clear twice to avoid leftovers in PHP 7.4 with ext-xdebug and code coverage turned on
 
+        /** @var Deferred $deferred */
```

Comment:
This seems rather obvious to me or am I missing something?

Reply:
Fair enough, change doesn't to any harm, is unrelated to this PR and was done to make Eclipse IDE happy only. Removed any unrelated changes as suggested, PR is ready for review again :shipit:

Reply:
Ow I'm not opposed to them, I just found it weird a so obvious type has to be defined.

Reply:
Agree, will take a look at this again and file a separate PR if it makes sense :+1: